WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/utils/SiteModelProvider.kt

```diff
@@ -0,0 +1,46 @@
+package org.wordpress.android.ui.stats.refresh.utils
+
+import android.arch.lifecycle.LiveData
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.fluxc.store.SiteStore.OnSiteChanged
+import org.wordpress.android.viewmodel.SingleLiveEvent
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class SiteModelProvider
+@Inject constructor(private val siteStore: SiteStore, private val dispatcher: Dispatcher) {
+    lateinit var siteModel: SiteModel
+    private val mutableSiteChanged = SingleLiveEvent<OnSiteChanged>()
+    val siteChanged: LiveData<OnSiteChanged> = mutableSiteChanged
+
+    fun start(site: SiteModel) {
+        dispatcher.register(this)
+        this.siteModel = site
+    }
+
+    fun clear() {
+        if (mutableSiteChanged.value != null) {
+            mutableSiteChanged.value = null
+        }
+    }
+
+    fun stop() {
+        dispatcher.unregister(this)
+    }
+
+    fun hasLoadedSite(): Boolean = siteModel.siteId != 0L
+
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    fun onSiteChanged(event: OnSiteChanged) {
+        val site = siteStore.getSiteByLocalId(siteModel.id)
```

Comment:
This looks a bit cumbersome. When updating a the site provider when the site changes you must already know what it changed to :). Would it be possible to update the `OnSiteChanged` to include the new site value? That way you it won't be necessary to call `start(site)`. Also, that way the dispatcher registration could happen in the constructor.

Reply:
You're totally right and I think it should be done this way. I think it's out of scope for this PR. At least we're containing this logic in one place. I think it might make sense to rename this `SiteModelProvider` to something like `StatsSiteProvider` because it shouldn't be used anywhere else right now. I think adding a full reusable `SiteModelProvider` should be a separate task. Let me know what you think :-)